lib/bolt_server/config.rb

```diff
@@ -1,118 +1,31 @@
 # frozen_string_literal: true
 
 require 'hocon'
+require 'service_config/base_config'
 require 'bolt/error'
 
 module BoltServer
-  class Config
-    CONFIG_KEYS = ['host', 'port', 'ssl-cert', 'ssl-key', 'ssl-ca-cert',
-                   'ssl-cipher-suites', 'loglevel', 'logfile', 'whitelist', 'concurrency',
-                   'cache-dir', 'file-server-conn-timeout', 'file-server-uri'].freeze
-
-    ENV_KEYS = ['ssl-cert', 'ssl-key', 'ssl-ca-cert', 'loglevel',
-                'concurrency', 'file-server-conn-timeout', 'file-server-uri'].freeze
-
-    DEFAULTS = {
-      'host' => '127.0.0.1',
-      'port' => 62658,
-      'ssl-cipher-suites' => ['ECDHE-ECDSA-AES256-GCM-SHA384',
-                              'ECDHE-RSA-AES256-GCM-SHA384',
-                              'ECDHE-ECDSA-CHACHA20-POLY1305',
-                              'ECDHE-RSA-CHACHA20-POLY1305',
-                              'ECDHE-ECDSA-AES128-GCM-SHA256',
-                              'ECDHE-RSA-AES128-GCM-SHA256',
-                              'ECDHE-ECDSA-AES256-SHA384',
-                              'ECDHE-RSA-AES256-SHA384',
-                              'ECDHE-ECDSA-AES128-SHA256',
-                              'ECDHE-RSA-AES128-SHA256'],
-      'loglevel' => 'notice',
-      'concurrency' => 100,
-      'cache-dir' => "/opt/puppetlabs/server/data/bolt-server/cache",
-      'file-server-conn-timeout' => 120
-    }.freeze
-
-    CONFIG_KEYS.each do |key|
-      define_method(key.tr('-', '_').to_sym) do
-        @data[key]
-      end
-    end
-
-    def initialize(config = nil)
-      @data = DEFAULTS.clone
-      @data = @data.merge(config.select { |key, _| CONFIG_KEYS.include?(key) }) if config
-      @config_path = nil
+  class Config < BoltServer::BaseConfig
+    def self.config_keys
```

Comment:
This will be simpler if it's not a class method and just users super probably. True for the other shared functions too.

```
def config_keys
  super << 'concurrency'
end
```